WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalBuiltInConnectingFailed.swift

```diff
@@ -55,6 +55,20 @@ final class CardPresentModalBuiltInConnectingFailed: CardPresentPaymentsModalVie
     func didTapAuxiliaryButton(in viewController: UIViewController?) { }
 }
 
+extension CardPresentModalBuiltInConnectingFailed: ReaderConnectionUnderlyingErrorDisplaying {
+    func errorDescription(underlyingError: CardReaderServiceUnderlyingError) -> String? {
+        switch underlyingError {
+        case .internalServiceError:
+            return NSLocalizedString(
+                "Sorry, we could not start Tap to Pay on iPhone. Please check your connection and try again.",
+                comment: "Error message when the built-in reader connection experiences an unexpected internal service error."
+            )
+        default:
+            return underlyingError.errorDescription
+        }
+    }
+}
```

Comment:
Not a blocker: I see we're repeating the same implementation in `CardPresentModalBuiltInConnectingFailed` and `CardPresentModalBuiltInConnectingFailedNonRetryable`, perhaps we can provide a default implementation directly in the protocol?

Reply:
I chose not to do that in this case, because I want to allow different alerts to conform to that protocol but provide differently specified messages, based on their own context. I.e. a `CardPresentModalBuiltInPaymentFailed` modal could conform to it and provide payment-specific error language.